mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -306,44 +309,46 @@ func resourceMongoDBAtlasOnlineArchiveImportState(ctx context.Context, d *schema
 
 		projectID = parts[0]
 		clusterName = strings.Join(parts[1:len(parts)-2], "")
-		atlasID = parts[len(parts)-1]
+		archiveID = parts[len(parts)-1]
 	} else {
-		projectID, clusterName, atlasID = parts[0], parts[1], parts[2]
+		projectID, clusterName, archiveID = parts[0], parts[1], parts[2]
 	}
 
-	outOnlineArchive, _, err := conn.OnlineArchives.Get(ctx, projectID, clusterName, atlasID)
+	outOnlineArchive, _, err := connV2.OnlineArchiveApi.GetOnlineArchive(ctx, projectID, archiveID, clusterName).Execute()
 
 	if err != nil {
-		return nil, fmt.Errorf("could not import Online Archive %s in project %s, error %s", atlasID, projectID, err.Error())
+		return nil, fmt.Errorf("could not import Online Archive %s in project %s, error %s", archiveID, projectID, err.Error())
 	}
 
 	// soft error, because after the import will be a read execution
 	if err := d.Set("project_id", projectID); err != nil {
-		log.Printf("error setting project id %s for Online Archive id: %s", err, atlasID)
+		log.Printf("error setting project id %s for Online Archive id: %s", err, archiveID)
 	}
 
-	mapValues := fromOnlineArchiveToMapInCreate(outOnlineArchive)
+	mapValues := fromOnlineArchiveToMap(outOnlineArchive)
 	for key, val := range mapValues {
 		if err := d.Set(key, val); err != nil {
-			return nil, fmt.Errorf("error MongoDB Atlas Online Archive with id %s, read error: %w", atlasID, err)
+			return nil, fmt.Errorf("error MongoDB Atlas Online Archive with id %s, read error: %w", archiveID, err)
 		}
 	}
 
 	d.SetId(encodeStateID(map[string]string{
-		"archive_id":   outOnlineArchive.ID,
-		"cluster_name": outOnlineArchive.ClusterName,
+		"archive_id":   util.SafeString(outOnlineArchive.Id),
+		"cluster_name": util.SafeString(outOnlineArchive.ClusterName),
 		"project_id":   projectID,
 	}))
 
 	return []*schema.ResourceData{d}, nil
 }
 
-func mapToArchivePayload(d *schema.ResourceData) matlas.OnlineArchive {
+func mapToArchivePayload(d *schema.ResourceData) admin.BackupOnlineArchiveCreate {
 	// shared input
-	requestInput := matlas.OnlineArchive{
-		DBName:         d.Get("db_name").(string),
-		CollName:       d.Get("coll_name").(string),
-		CollectionType: d.Get("collection_type").(string),
+	requestInput := admin.BackupOnlineArchiveCreate{
+		DbName:   d.Get("db_name").(string),
```

Comment:
type conversions are normally recommended with error var

Reply:
it is a valid point, I would however consider that value is obtained from terraform where resource schema is being enforced.